File: backend/user/golang/routes/user_route.go
package route

import (
	controller "project-user/controllers"
	middleware "project-user/middlewares"
	util "project-user/utils"

	"github.com/labstack/echo/v4"
)

func UserRoute(e *echo.Echo, redisUtil util.RedisUtil, controller controller.UserController) {
	e.POST("/api/v1/user/register", controller.Register)
	e.POST("/api/v1/user/login", controller.Login)
	e.POST("/api/v1/user/logout", controller.Logout, middleware.Authenticate(redisUtil))
}

File: backend/user/nodejs/src/routes/user-route.js
import express from "express";
import userController from "../controllers/user-controller.js";
import authenticationMiddleware from "../middlewares/authentication-middleware.js";

const userRouter = new express.Router()
userRouter.post("/api/v1/user/register", userController.register)
userRouter.post("/api/v1/user/login", authenticationMiddleware.getSessionId, userController.login)
userRouter.post("/api/v1/user/logout", authenticationMiddleware.authenticate, userController.logout)

export {
    userRouter
}